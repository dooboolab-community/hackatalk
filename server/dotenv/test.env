JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
