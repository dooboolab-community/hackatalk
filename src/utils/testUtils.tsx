import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
