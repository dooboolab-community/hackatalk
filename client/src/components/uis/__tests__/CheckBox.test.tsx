import 'react-native';

import * as React from 'react';

import {createTestElement, createTestProps} from '../../../../test/testUtils';

import Shared from '../CheckBox';
import {render} from '@testing-library/react-native';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: React.ReactElement;

describe('Rendering', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Shared {...props} />);
  });

  it('renders without crashing', () => {
    const json = render(component).toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });
});
