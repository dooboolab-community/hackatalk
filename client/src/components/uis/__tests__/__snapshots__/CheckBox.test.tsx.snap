// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders without crashing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "flex-start",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomColor": "#707070",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderBottomWidth": 2,
              "borderLeftColor": "#707070",
              "borderLeftWidth": 2,
              "borderRightColor": "#707070",
              "borderRightWidth": 2,
              "borderTopColor": "#707070",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "borderTopWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
            Array [
              undefined,
              Object {
                "backgroundColor": "white",
                "borderColor": "#000",
                "borderWidth": 2,
              },
            ],
          ]
        }
      />
    </View>
    <Text
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "fontSize": 14,
            "marginLeft": 8,
          },
          Object {
            "color": "#000",
          },
        ]
      }
      testID="undefined-nofeed"
    >
      
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
